File: Go/oddevensort/oddevensort_250000.go
package main

import (
	"fmt"
)

func OddEvenSort(arr []int) []int {
	n := len(arr)
	sorted := false

	for !sorted {
		sorted = true

		for i := 1; i < n-1; i += 2 {
			if arr[i] > arr[i+1] {
				arr[i], arr[i+1] = arr[i+1], arr[i]
				sorted = false
			}
		}

		for i := 0; i < n-1; i += 2 {
			if arr[i] > arr[i+1] {
				arr[i], arr[i+1] = arr[i+1], arr[i]
				sorted = false
			}
		}
	}

	return arr
}

func main() {
	debug := false
	OddEvenSort(arr)
	if debug {
		fmt.Println("Sorted array:", arr)
	}
}

File: Rust/oddevensort/oddevensort_250000.rs
fn odd_even_sort<T: PartialOrd>(arr: &mut [T]) {
    let len = arr.len();
    let mut sorted = false;
    while !sorted {
        sorted = true;
        for i in (0..len - 1).step_by(2) {
            if arr[i] > arr[i + 1] {
                arr.swap(i, i + 1);
                sorted = false;
            }
        }
        for i in (1..len - 1).step_by(2) {
            if arr[i] > arr[i + 1] {
                arr.swap(i, i + 1);
                sorted = false;
            }
        }
    }
}

fn main() {
    let debug: bool = false;
    odd_even_sort(&mut arr);
    if debug {
        println!("{:?}", arr);
    }    
}